src/librustc_mir/const_eval.rs

```diff
@@ -655,7 +653,7 @@ pub fn const_eval_raw_provider<'tcx>(
 
     let res = ecx.load_mir(cid.instance.def, cid.promoted);
     res.and_then(
-        |body| eval_body_using_ecx(&mut ecx, cid, body, key.param_env)
```

Comment:
This was redundant, as `ecx.param_env` is already `key.param_env`. So I removed the parameter here and propagated that to the interner.